scripts/SANS/sans/gui_logic/models/state_gui_model.py

```diff
@@ -888,7 +889,7 @@ def q_1d_rebin_string(self):
 
     @q_1d_rebin_string.setter
     def q_1d_rebin_string(self, value):
-        self._set_q_1d_limits(rebin_string=value)
+        self._set_q_1d_limits(rebin_string=check_for_bytes(value))
```

Comment:
Would a `str(value)` not suffice here?

Reply:
Doesn't `str(b"aString")` just return `"b'aString'")`? If it's of type `bytes` we need to decode it

Reply:
Doesn't `str(b"aString")` just return `"b'aString'")`? If it's of type `bytes` we need to decode it

Reply:
I also thought that this was the case?

Reply:
Yes. My apologies, I misread the `str()` docs and thought it default decoded with `utf-8`. We could use [`mantid.py3compat.ensure_str`](https://github.com/benjaminp/six/blob/master/six.py) ?

Reply:
That looks like a better solution